spur/core/base.py

```diff
@@ -179,6 +201,81 @@ def __init__(self, model, uid, store: Store) -> None:
         super().__init__(model, uid)
 
 
+class SpurRequest(Request):

```

Comment:
Should we also include an agent setter method in this class? We may end up wanting to assign it later on.

```python
@agent.setter
def agent(self, agent):
    self._agent = agent
```

Reply:
The agent is a required parameter in the constructor of `SpurRequest` and must be passed in when creating the request object. I'm not sure if it would make sense to modify the agent later on, since the idea was to bind the request to the agent that made it.

Reply:
Okedoke - makes sense we can leave as is, it's easy to implement later if needed.